City-of-Helsinki/kukkuu-ui
# KK-504 | Add support for languages spoken at home

## Description

Adds support for the `languagesSpokenAtHome` field. Guardians can now control this information in their profiles through a multiselect combobox component. As a part of this feature, I helped the HDS team implement the `Combobox` component into HDS.

## Context

After these change users can provide information about the languages that are spoken in their homes. The Culture Kids maintainers are then able to use this data to offer events that take user languages into account better.

[KK-504](https://helsinkisolutionoffice.atlassian.net/browse/KK-504)
Closes #274 

## How Has This Been Tested?

I added tests, but they don't actually cover the languages spoken at home field. In the implementation I chose to entangle data with the component for choosing languages. This means that other components using it don't have to load the data, but it also means that testing the component itself requires mocking the API. I'm not entirely sure of how strong of a sign it's to manage the data differently, but for now I did not make changes.

This functionality will be easier to cover by using `e2e` tests. It would be included in
* https://helsinkisolutionoffice.atlassian.net/browse/KK-591
* https://helsinkisolutionoffice.atlassian.net/browse/KK-600

## Manual Testing Instructions for Reviewers

Without logging in with a Tunnistamo account that's not registered to Kukkuu

1. Open home page
2. Click register
3. Fill short form
4. Authenticate in Tunnistamo
5. Fill long form
    - Expect to be able to control languages spoken at home
    - Feel free to test the combobox in the way you feel appropriate. There are many usecases for it of which many can't likely be fixed in this PR
6. Submit registration form
7. Click Edit information
8. Expect to see languages spoken at home filled with the values you chose previously
9. Expect to be able to edit those values
